mlir/test/Gradient/BufferizationTest.mlir

```diff
@@ -33,43 +33,42 @@ func.func @adjoint(%arg0: f64, %arg1: index) {
 func.func private @circuit2(%arg0: f64)
 
 // CHECK-LABEL: @backprop
-func.func @backprop(%arg0: f64, %arg1: memref<?xf64>) {
+func.func @backprop(%arg0: f64, %arg1: tensor<?xf64>) {
 
-    // CHECK:   [[alloc:%.+]] = memref.alloc() : memref<f64>
-    // CHECK:   gradient.backprop @circuit2({{.*}}) qjacobian(%arg1 : memref<?xf64>) in([[alloc]] : memref<f64>) {diffArgIndices = dense<0> : tensor<1xindex>} : (f64) -> ()
-    %grad = gradient.backprop @circuit2(%arg0) qjacobian(%arg1: memref<?xf64>) {diffArgIndices = dense<0> : tensor<1xindex>}: (f64) -> tensor<f64>
+    // CHECK:   [[dim:%.+]] = memref.dim
+    // CHECK:   [[calleeRes:%.+]] = memref.alloc([[dim]]) : memref<?xf64>
+    // CHECK:   gradient.backprop @circuit2({{%.+}}) callee_out([[calleeRes]] : memref<?xf64>) cotangents({{%.+}} : memref<?xf64>) {diffArgIndices = dense<0> : tensor<1xindex>} : (f64) -> f64
+    %grad = gradient.backprop @circuit2(%arg0) cotangents(%arg1: tensor<?xf64>) {diffArgIndices = dense<0> : tensor<1xindex>}: (f64) -> f64
     return
 }
 
 // -----
 
-func.func private @circuit3(%arg0: tensor<?x2xf64>)
+func.func private @circuit3(%arg0: tensor<3x2xf64>)
 
 // CHECK-LABEL: @backprop2
-func.func @backprop2(%arg0: tensor<?x2xf64>, %arg1: memref<?xf64>) {
+func.func @backprop2(%arg0: tensor<3x2xf64>, %arg1: tensor<?xf64>) {
```

Comment:
This test and the next seem to avoid the dynamic dimension, was there a specific reason for that or did it just come about from removing and re-adding the tests?

Reply:
Updated back